usaspending_api/etl/elasticsearch_loader_helpers/load_data.py

```diff
@@ -310,170 +45,28 @@ def streaming_post_to_es(
-        client.indices.delete_alias(index, "_all")
-    if load_type == "awards":
-        alias_patterns = settings.ES_AWARDS_QUERY_ALIAS_PREFIX + "*"
-    else:
-        alias_patterns = settings.ES_TRANSACTIONS_QUERY_ALIAS_PREFIX + "*"
-    old_indexes = []
-
+def create_index(index, client):
     try:
-        old_indexes = list(client.indices.get_alias("*", alias_patterns).keys())
-        for old_index in old_indexes:
-            client.indices.delete_alias(old_index, "_all")
-            logger.info(format_log(f"Removing aliases from '{old_index}'", process="ES Alias Drop"))
+        does_index_exist = client.indices.exists(index)
     except Exception:
-        logger.exception(format_log(f"No aliases found for {alias_patterns}", process="ES Alias Drop"))
-
-    create_aliases(client, index, load_type=load_type)
-
-    try:
-        if old_indexes:
-            client.indices.delete(index=old_indexes, ignore_unavailable=False)
-            logger.info(format_log(f"Deleted index(es) '{old_indexes}'", process="ES Alias Drop"))
-    except Exception:
-        logger.exception(format_log(f"Unable to delete indexes: {old_indexes}", process="ES Alias Drop"))
-
-
-def post_to_elasticsearch(client, job, config, chunksize=250000):
-    logger.info(format_log(f"Populating ES Index '{job.index}'", job=job.name, process="ES Index"))
-    start = perf_counter()
-    try:
-        does_index_exist = client.indices.exists(job.index)
-    except Exception as e:
-        print(e)
+        logger.exception("Unable to query cluster for indices")
         raise SystemExit(1)
     if not does_index_exist:
-        logger.info(format_log(f"Creating index '{job.index}'", job=job.name, process="ES Index"))
-        client.indices.create(index=job.index)
-        client.indices.refresh(job.index)
-
-    csv_generator = csv_chunk_gen(job.csv, chunksize, job.name, config["load_type"])
-    for count, chunk in enumerate(csv_generator):
-        if len(chunk) == 0:
-            logger.info(format_log(f"No documents to add/delete for chunk #{count}", job=job.name, process="ES Index"))
-            continue
-
-        # Only delete before adding/inserting/indexing new docs on incremental loads, not full reindexes
-        is_incremental = config["is_incremental_load"] and str(config["is_incremental_load"]).lower() == "true"
-
-        iteration = perf_counter()
-        current_rows = f"({count * chunksize + 1:,}-{count * chunksize + len(chunk):,})"
-        logger.info(
-            format_log(f"ES Stream #{count} rows [{current_rows}/{job.count:,}]", job=job.name, process="ES Index")
-        )
-        streaming_post_to_es(
-            client, chunk, job.index, config["load_type"], job.name, delete_before_index=is_incremental
-        )
-        if is_incremental:
-            # refresh_interval is off during incremental loads.
-            # Manually refresh after delete + insert complete for search consistency
-            client.indices.refresh(job.index)
```

Comment:
I'm not seeing the importance of adding a manual refresh after each partition is loaded. Seems like it would add overhead with no actual benefit.

Reply:
When we do an incremental load that is taking hours, those are hours of inconsistency between postgres RDS and elasticsearch data.

Just because we have a cache, doesn't mean everything is cached -- only the stuff that was part of an API response from the prior day. If new data comes in, or if a user requests data from the API in a different way, or if they run a download ... the cache won't be in play and there is the possibility of inconsistent data.

Closing that gap of inconsistency wherever we can is important.

Reply:
The Elasticsearch data will be inconsistent from the moment the materialized views are refreshed to the moment the ES ETL completes. I don't see any gaps to close other than the total ES ETL runtime